gpflow/posterior.py

```diff
@@ -235,31 +512,67 @@ def predict_f(
             cov_shape = (K, N, N) if full_cov else (N, K, K)
         cov = tf.reshape(cov, cov_shape)
 
-        return mean + self.mean_function(Xnew), cov
-
+        return mean, cov
 
-def _get_kernels(Xnew, inducing_variable, kernel, full_cov, full_output_cov):
+    def _conditional_fused(
+        self, Xnew, full_cov: bool = False, full_output_cov: bool = False
+    ) -> MeanAndVariance:
+        Kmm = covariances.Kuu(
+            self.inducing_variable, self.kernel, jitter=default_jitter()
+        )  # [M, L, M, L]
+        Kmn = covariances.Kuf(self.inducing_variable, self.kernel, Xnew)  # [M, L, N, P]
+        Knn = self.kernel(
+            Xnew, full_cov=full_cov, full_output_cov=full_output_cov
```

Comment:
Should the `full_output_cov` argument be removed?

Reply:
I don't think so; it changes the shape of Knn....